File: client/js/mob.js

define(['character', 'player'], function(Character) {
    
    var Mob = Character.extend({
        init: function(id, kind) {
            this._super(id, kind);

            this.targetable = true;
            this.aggroRange = 1;
            this.isAggressive = true;
        },
        
        isHostile: function(entity) {
            // @TODO: implement!
            return (entity instanceof Player);
        },

    	die: function() {
            // Keep track of where mobs die in order to spawn their dropped items
            // at the right position later.
            globalGame.deathpositions[this.id] = {x: this.gridX, y: this.gridY};

            this._super();
    	},
    });
    
    return Mob;
});

File: client/dart/mob.dart
library mob;

import "character.dart";
import "entity.dart";
import "game.dart";
import "player.dart";
import "lib/gametypes.dart";

class Mob extends Character {

  bool isAggressive = true;
  bool targetable = true;

  Mob(int id, EntityKind kind): super(id, kind);

  bool isHostile(Entity entity) => (entity is Player);

  void die() {
    Game.registerEntityDeathPosition(this);

    super.die();
  }
}
